packages/jasmine/src/coverage_runner.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2017 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const childProcess = require('child_process');
+
+/**
+ * This is designed to collect the coverage of one target, since in nodejs
+ * and using NODE_V8_COVERAGE it may produce more than one coverage file, however bazel expects
+ * there to be only one lcov file. So this collects up the v8 coverage json's merges them and
+ * converts them to lcov for bazel to pick up later.
+ * TODO: The functionality in this script could be put into _lcov_merger then it would apply to
+ * any tool reporting coverage not just jasmine
+ */
+async function main() {
+  const result = childProcess.spawnSync(
+      process.execPath,
+      [...process.execArgv, __dirname + '/jasmine_runner.js', ...process.argv.slice(2)],
+      {stdio: 'inherit'});
+
+  const exitCode = result.status;
+
+  // 0 indicates success
+  // so if it's anything tthen exit here since bazel only collects coverage on success
+  if (exitCode !== 0) {
+    process.exit(exitCode);
+  }
+
+  if (!coverageEnabled) {
+    process.exit(exitCode);
+  }
+
+  const coverageDir = process.env.COVERAGE_DIR;
+  const outputFile = process.env.COVERAGE_OUTPUT_FILE;
+  const sourceFileManifest = process.env.COVERAGE_MANIFEST;
+
+  const instrumentedSourceFiles = fs.readFileSync(sourceFileManifest).toString('utf8').split('\n');
+
+  // c8 will name the output report file lcov.info
+  // so we give it a dir that it can write to
+  // later on we'll move and rename it into output_file as bazel expects
+  const tmpdir = process.env['TEST_TMPDIR'];
+  const c8OutputDir = path.join(tmpdir, crypto.randomBytes(4).toString('hex'));
+  fs.mkdirSync(c8OutputDir);
+
+  const includes = instrumentedSourceFiles
+                       // we can only cover js files
+                       .filter(f => ['.js', '.jsx', '.ts', '.tsx'].includes(path.extname(f)))
```

Comment:
`.mjs`  as well?

Seems like this in an ideal world this list would be configurable somewhere.

Reply:
Agreed, this is defined in serveral different places now, would be great to have a single definition

Reply:
Hmmm in the code below we convert all ext's to .js but .mjs is a valid ext, will need to make that more complex

Reply:
im not actually sure how to handle this now, I've added a few comments in the code here but it's really this part:
```
// TODO: how do we know what source files produce .mjs?
```